connector/solarwindsentityconnector/benchmark/data_generator.go

```diff
@@ -0,0 +1,285 @@
+func genMetricsFromConfig(b *testing.B, cfg *solarwindsentityconnector.Config, count int, invalidRatio float32, multiple bool) []pmetric.Metrics {
+	b.Helper()
+
+	var uniqueMetrics []pmetric.Metrics
+	for _, e := range cfg.Schema.Events.Entities {
+		if e.Context != ottlmetric.ContextName {
+			continue
+		}
+		metric := buildEntityMetric(b, cfg, e.Type)
+		uniqueMetrics = append(uniqueMetrics, metric)
+	}
+
+	for _, r := range cfg.Schema.Events.Relationships {
+		if r.Context != ottlmetric.ContextName {
+			continue
+		}
+		metric := buildRelationshipMetric(b, cfg, r.Type, r.Source, r.Destination)
+		uniqueMetrics = append(uniqueMetrics, metric)
+	}
+
+	invalidCount := int(float32(count) * invalidRatio)
+	validCount := count - invalidCount
+
+	finalMetrics := make([]pmetric.Metrics, 0, count)
+
+	// If multiple flag is true, will build a multiple metric
+	// that contains all unique metrics repeated as provided by the count.
+	if multiple {
+		finalMetrics = append(finalMetrics, buildMultipleMetric(b, count, uniqueMetrics))
+	} else {
+		for i := 0; i < validCount; i++ {
+			finalMetrics = append(finalMetrics, uniqueMetrics[i%len(uniqueMetrics)])
+		}
+	}
+
+	for i := 0; i < invalidCount; i++ {
+		finalMetrics = append(finalMetrics, buildInvalidMetric(b, i))
+	}
+
+	return finalMetrics
+}
+
+func buildEmptyMetric(b *testing.B) pmetric.Metrics {
+	b.Helper()
+
+	metric := pmetric.NewMetrics()
+	rm := metric.ResourceMetrics().AppendEmpty()
+
+	sm := rm.ScopeMetrics().AppendEmpty()
+	sm.Scope().SetName("benchmark")
+
+	m := sm.Metrics().AppendEmpty()
+	m.SetName("dummy_metric")
+	m.SetEmptyGauge()
+
+	return metric
+}
+
+func buildEntityMetric(b *testing.B, cfg *solarwindsentityconnector.Config, entityType string) pmetric.Metrics {
+	b.Helper()
+
+	metric := buildEmptyMetric(b)
+	attrs := metric.ResourceMetrics().At(0).Resource().Attributes()
+
+	for _, ent := range cfg.Schema.Entities {
+		if ent.Type != entityType {
+			continue
+		}
+		for _, id := range ent.IDs {
+			attrs.PutStr(id, fmt.Sprintf("%s.%s", id, ent.Type))
+		}
+		break
+	}
+
+	return metric
+}
+
+func buildRelationshipMetric(b *testing.B, cfg *solarwindsentityconnector.Config, relType, sourceType, destType string) pmetric.Metrics {
+	b.Helper()
+
+	metric := buildEmptyMetric(b)
+	attrs := metric.ResourceMetrics().At(0).Resource().Attributes()
+
+	for _, ent := range cfg.Schema.Entities {
+		if ent.Type == sourceType {
+			for _, id := range ent.IDs {
+				attrs.PutStr(fmt.Sprintf("src.%s", id), fmt.Sprintf("%s.%s.%s", id, ent.Type, relType))
+			}
+		}
+
+		if ent.Type == destType {
+			for _, id := range ent.IDs {
+				attrs.PutStr(fmt.Sprintf("dst.%s", id), fmt.Sprintf("%s.%s.%s", id, ent.Type, relType))
+			}
+		}
+	}
+
+	return metric
+}
+
+func buildInvalidMetric(b *testing.B, i int) pmetric.Metrics {
+	b.Helper()
+
+	metric := buildEmptyMetric(b)
+	attrs := metric.ResourceMetrics().At(0).Resource().Attributes()
+	attrs.PutStr("InValid", fmt.Sprintf("%d", i))
```

Comment:
Could we add more attributes? Real metric/log will have more resource attributes (my rough guess is 5-10 attributes), so it would be nice to see how it behaves with more attributes (the ottl parsing and validation). For one processing 10 attributes is nothing, but for 1M it could make a difference.

```suggestion
        for i := 0; i < 10; i++ {
        	attrs.PutStr(fmt.Sprintf("invalid-%d", i), fmt.Sprintf("%d", i))
        }
```